### PR TITLE
BAEL-4077 JPA Entity Equality

### PR DESCRIPTION
**Pull request by andrebrowne**

### CODE REVIEW

**Comment by andrebrowne** [comment]
@glmartin

**Comment by andrebrowne** [comment]
@glmartin Just a ping to make sure you saw this

File: persistence-modules/java-jpa-3/src/main/java/com/baeldung/jpa/equality/EqualByBusinessKey.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (obj instanceof EqualByBusinessKey)
+            if (((EqualByBusinessKey) obj).getEmail() == getEmail())
+                return true;
```
Comment (KevinGilmore) [inline persistence-modules/java-jpa-3/src/main/java/com/baeldung/jpa/equality/EqualByBusinessKey.java:46]: Please use braces in all if/else blocks, even on one-liners, as this is considered a best practice.

File: persistence-modules/java-jpa-3/src/main/java/com/baeldung/jpa/equality/EqualById.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (obj instanceof EqualById)
+            return ((EqualById) obj).getId() == getId();
```
Comment (KevinGilmore) [inline persistence-modules/java-jpa-3/src/main/java/com/baeldung/jpa/equality/EqualById.java:49]: braces

File: persistence-modules/java-jpa-3/src/main/java/com/baeldung/jpa/equality/EqualByJavaDefault.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        return email;
+    }
+
+    public void setEmail(String email) {
+        this.email = email;
+    }
+
+    public Object clone() throws
+            CloneNotSupportedException
+    {
```
Comment (KevinGilmore) [inline persistence-modules/java-jpa-3/src/main/java/com/baeldung/jpa/equality/EqualByJavaDefault.java:35]: opening brace should be at end of a line but not on a line by itself

File: persistence-modules/java-jpa-3/src/main/java/com/baeldung/jpa/equality/EqualById.java (hunk 2 above)
Comment (KevinGilmore) [inline persistence-modules/java-jpa-3/src/main/java/com/baeldung/jpa/equality/EqualById.java:49]: For wrapper classes, the == operator would only return true for values in the range -127 to 127. Should use the equals method to compare the ids.

File: persistence-modules/java-jpa-3/src/main/java/com/baeldung/jpa/equality/EqualByBusinessKey.java (hunk 1 above)
Comment (andrebrowne) [reply persistence-modules/java-jpa-3/src/main/java/com/baeldung/jpa/equality/EqualByBusinessKey.java:46]: Fixed

File: persistence-modules/java-jpa-3/src/main/java/com/baeldung/jpa/equality/EqualById.java (hunk 2 above)
Comment (andrebrowne) [reply persistence-modules/java-jpa-3/src/main/java/com/baeldung/jpa/equality/EqualById.java:49]: Fixed

File: persistence-modules/java-jpa-3/src/main/java/com/baeldung/jpa/equality/EqualByJavaDefault.java (hunk 3 above)
Comment (andrebrowne) [reply persistence-modules/java-jpa-3/src/main/java/com/baeldung/jpa/equality/EqualByJavaDefault.java:35]: Fixed

File: persistence-modules/java-jpa-3/src/main/java/com/baeldung/jpa/equality/EqualById.java (hunk 2 above)
Comment (andrebrowne) [reply persistence-modules/java-jpa-3/src/main/java/com/baeldung/jpa/equality/EqualById.java:49]: Fixed

**Pull request merged**